App/View Controllers/Posts/PostsPageView.swift

```diff
@@ -251,41 +286,79 @@ final class PostsPageView: UIView {
         loadingViewContainer.frame = bounds
 
         let toolbarHeight = toolbar.sizeThatFits(bounds.size).height
+        let toolbarY: CGFloat
+        
+        if immersionModeEnabled && immersionModeState == .immersed {
+            // Position toolbar off-screen when in immersion mode
+            toolbarY = bounds.maxY
+        } else {
+            toolbarY = bounds.maxY - layoutMargins.bottom - toolbarHeight
+        }
+        
         toolbar.frame = CGRect(
-            x: bounds.minX,
-            y: bounds.maxY - layoutMargins.bottom - toolbarHeight,
-            width: bounds.width,
+            x: safeAreaInsets.left,
+            y: toolbarY,
+            width: bounds.width - safeAreaInsets.left - safeAreaInsets.right,
             height: toolbarHeight)
 
         let scrollView = renderView.scrollView
 
-        let refreshControlHeight = refreshControlContainer.layoutFittingCompressedHeight(targetWidth: bounds.width)
+        let refreshControlHeight = refreshControlContainer.layoutFittingCompressedHeight(targetWidth: bounds.width - safeAreaInsets.left - safeAreaInsets.right)
         refreshControlContainer.frame = CGRect(
-            x: bounds.minX,
+            x: safeAreaInsets.left,
             y: max(scrollView.contentSize.height, scrollView.bounds.height - layoutMargins.bottom),
-            width: bounds.width,
+            width: bounds.width - safeAreaInsets.left - safeAreaInsets.right,
             height: refreshControlHeight)
 
-        let topBarHeight = topBarContainer.layoutFittingCompressedHeight(targetWidth: bounds.width)
+        let topBarHeight = topBarContainer.layoutFittingCompressedHeight(targetWidth: bounds.width - safeAreaInsets.left - safeAreaInsets.right)
         topBarContainer.frame = CGRect(
-            x: bounds.minX,
+            x: safeAreaInsets.left,
             y: bounds.minY + layoutMargins.top,
-            width: bounds.width,
+            width: bounds.width - safeAreaInsets.left - safeAreaInsets.right,
             height: topBarHeight)
+        
+        // Position gradient view to cover only the top safe area
+        let gradientHeight: CGFloat = layoutMargins.top
+        safeAreaGradientView.frame = CGRect(
+            x: 0,
+            y: bounds.minY,
+            width: bounds.width,
+            height: gradientHeight)
+        
+        // Update gradient layer frame to match view bounds
+        if let gradientLayer = safeAreaGradientView.layer.sublayers?.first as? CAGradientLayer {
```

Comment:
This is better done as a UIView subclass that overrides layerClass and does something like
```swift
var gradientLayer: CAGradientLayer { layer as! CAGradientLayer }
```
then we don't have to root around for the gradient layer and set its frame. Also then it works with UIKit animation.